### ISSUE TITLE
DietPi-Software | Node.js: ARMv6 binaries are missing in latest release

### ISSUE THREAD

**Issue by axwax** [MichaIng/DietPi#2755]
### Solution
```
wget https://raw.githubusercontent.com/taaem/nodejs-linux-installer/master/node-install.sh
sed -i 's|nodejs.org/dist/latest/|nodejs.org/dist/latest-v11.x/|g' node-install.sh
chmod +x node-install.sh
./node-install.sh
rm node-install.sh
```
___________
### Creating a bug report/issue
Despite Node-Red seemingly installing fine it doesn't appear to be installed where its service thinks it is (/usr/local/bin/node-red). Additionally, the node-red command can't be found.

#### Required Information
- DietPi version | 6.22.3 (master/Fourdee)
- Distro version | Stretch 9.9
- Kernel version | Linux homecontrol 4.14.98+ #1200 Tue Feb 12 20:11:02 GMT 2019 armv6l GNU/Linux
- SBC device | RPi Zero W (armv6l)
- Power supply used | 5V, 2.1A
- SDcard used | Kingston SDCS 16GB

#### Additional Information (if applicable)
- Software title | Node Red
- Was the software title installed freshly or updated/migrated? installed freshly
- Can this issue be replicated on a fresh installation of DietPi? it was a fresh install, but upgraded from 6.16 (I think)
- Bug report 12cecaf2-19ae-476c-9625-7163da6f0081

#### Steps to reproduce

1. upgrade DietPi 6.16 to 6.22.3
2. in dietpi-software choose mosquitto, node-red, influxdb, grafana-server
3. run systemctl status node-red.service
4. run node-red

#### Expected behaviour

    Loaded: loaded
    Active: active
    ...

```
node-red
Welcome to Node-RED
===================
```

#### Actual behaviour

```
● node-red.service - Node-Red (DietPi)
Loaded: loaded (/etc/systemd/system/node-red.service; disabled; vendor preset: enabled)
Active: failed (Result: exit-code) since Thu 2019-05-02 15:11:03 BST; 14min ago
Process: 9716 ExecStart=/usr/local/bin/node-red -u /mnt/dietpi_userdata/node-red (code=exited, status=203/EXEC)
Main PID: 9716 (code=exited, status=203/EXEC)

May 02 15:11:03 homecontrol systemd[1]: Started Node-Red (DietPi).
May 02 15:11:03 homecontrol systemd[9716]: node-red.service: Failed at step EXEC spawning /usr/local/bin/node-red: No such file or direc
tory
May 02 15:11:03 homecontrol systemd[1]: node-red.service: Main process exited, code=exited, status=203/EXEC
May 02 15:11:03 homecontrol systemd[1]: node-red.service: Unit entered failed state.
May 02 15:11:03 homecontrol systemd[1]: node-red.service: Failed with result 'exit-code'.
```

```
node-red
-bash: node-red: command not found
```

#### Extra details
Other than the original install file being outdated (I think it was 6.16) it was a vanilla install, which I then upgraded to the current version before attempting any software install.

**Comment by MichaIng** [comment]
@axwax
Many thanks for your report.

Hmm, so the Node-RED install must have failed. Any error messages you remember from DietPi-Software run?

Please try to reinstall/verify manually:
`npm i -g --unsafe-perm node-red` 

The current RPi image btw is v6.22. However since it is updated on first boot automatically, it should not matter.

**Comment by axwax** [comment]
Wow, thanks for the super quick response!
I just realised the problem seems to be with the node.js installation:
While it is ticked in Software Additional there is neither a node nor an npm command.

I just did a `dietpi-software  reinstall 9`
and the installer fails:
```
2019-05-02 21:08:32 (619 KB/s) - ‘node_install.sh’ saved [1943/1943]

Node Linux Installer by www.github.com/taaem
Running as Root User
Get Latest Version Number...
Done
Downloading latest stable Version ...
Done
Installing...

gzip: stdin: not in gzip format
tar: Child returned status 1
tar: Error is not recoverable: exiting now
ERROR: Couldn't extract tar
```
I did choose the lightweight 7zip option when prompted during the upgrade process to 6.22.3 - could that be the problem?

**Comment by MichaIng** [comment]
@axwax 
I just tested and on x86 VM it succeeds.

> I did choose the lightweight 7zip option when prompted during the upgrade process to 6.22.3 - could that be the problem?

Nope, the error is about `gzip`, not `7zip`. Hum let me check the Node install code.

Here it looks like it should:
```
node_install.sh                         100%[===============================================================================>]   1.90K  --.-KB/s    in 0s

2019-05-02 21:43:23 (28.2 MB/s) - ‘node_install.sh’ saved [1943/1943]

Node Linux Installer by www.github.com/taaem
Running as Root User
Get Latest Version Number...
Done
Downloading latest stable Version node-v12.1.0-linux-x64.tar.gz...
Done
Installing...
Finished installing!
```

Node installer: https://raw.githubusercontent.com/taaem/nodejs-linux-installer/master/node-install.sh

... damn, there is no armv6l binary for the latest release: https://nodejs.org/dist/latest/

To compare, the lastest Node v11 release has armv6l binaries: https://nodejs.org/dist/latest-v11.x/

We are not the only ones who face this: https://github.com/taaem/nodejs-linux-installer/issues/9

**Comment by MichaIng** [comment]
See my comments in the Node GitHub issue. Looking at the download page it really seems to be planned/wanted that latest Node does not support ARMv6 anymore, thus does not work on many RPi anymore.

I really hope they rething this, however until then, and as the installer does not handle this gracefully, we have to manually choose Node 11 for ARMv6 then...

Damn thing, this means we have to revert to the Node APT repo I think as the installer does not ship ANY possibility to install any other than the latest version...

**Comment by MichaIng** [comment]
@axwax 
Okay found a solution that is not too complicated. We can fix the installer easily when downloaded. Please try:
```
wget https://raw.githubusercontent.com/taaem/nodejs-linux-installer/master/node-install.sh
sed -i 's|nodejs.org/dist/latest/|nodejs.org/dist/latest-v11.x/|g' node-install.sh
chmod +x node-install.sh
./node-install.sh
rm node-install.sh
```

Fixed with: https://github.com/MichaIng/DietPi/commit/dc09158a26fa602ab9313908e16f69e3065a7a2a
Changelog: https://github.com/MichaIng/DietPi/commit/dca651c7c225dd7c46155fe6a2d4475b756dcdeb
Ah now I got confused 😄: https://github.com/MichaIng/DietPi/commit/07ee4c49c679d78476851f899304d103c6e7545b

**Comment by axwax** [comment]
That's amazing! 
Ijust followed your above node.js install instructions, and got it all running.
I did have to  change `node_install` to `node-install` in lines 2-5 though.

I just looked at your commit, and I think you might have to check for that too: 
You changed `wget "$INSTALL_URL_ADDRESS" -O node_install.sh` to `wget "$INSTALL_URL_ADDRESS"`, so the downloaded file will be called `node-install.sh` making the following lines fail I think! Just reverting that line should be all that's needed...

Anyway, my Node Red has installed beautifully now - thanks for such a speedy fix, and I'm looking forward to 6.23! 😃
